lib/validate.js

```diff
@@ -158,20 +176,16 @@ const validateCategory = (
       continue;
     }
 
-    const opts = { ...options };
+    if (extractDecorator(definition) === 'List') {
+      value.forEach((listItem, index) => {
+        validate(listItem, `${prop}[${index}]`, definition);
+      });
 
-    let err;
-    if (definition.domain) {
-      opts.path = `${path}${prop}`;
-      err = ms.validate('domains', definition.domain, value, opts);
-    } else if (definition.category) {
-      opts.path = `${path}${prop}.`;
-      err = ms.validate('category', definition.category, value, opts);
-    }
-    if (err) {
-      errors.push(...err.errors);
+      continue;
```

Comment:
I'd prefer not to add `continue` here, so that `definition.validate` is checked.